jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/annotations/ResourceAnnotationQuickFix.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+* Copyright (c) 2021 IBM Corporation and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     IBM Corporation - initial API and implementation
+*******************************************************************************/
+
+
+
+
+package org.eclipse.lsp4jakarta.jdt.core.annotations;
+
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.dom.IBinding;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4jakarta.jdt.codeAction.IJavaCodeActionParticipant;
+import org.eclipse.lsp4jakarta.jdt.codeAction.JavaCodeActionContext;
+import org.eclipse.lsp4jakarta.jdt.codeAction.proposal.ChangeCorrectionProposal;
+import org.eclipse.lsp4jakarta.jdt.codeAction.proposal.ModifyAnnotationProposal;
+import org.eclipse.lsp4jakarta.jdt.codeAction.proposal.quickfix.InsertAnnotationMissingQuickFix;
+import org.eclipse.lsp4jakarta.jdt.core.annotations.AnnotationConstants;
+
+
+
+/**
+ * Quickfix for annotation Resource
+ * 1. Add missing name
+ * 2. Add missing type
+ * 
+ * @author Zijian Pei
+ *
+ */
+public class ResourceAnnotationQuickFix extends InsertAnnotationMissingQuickFix{
```

Comment:
See this comment on your other pull request: #196 (comment)
Create a common class for adding attributes to an annotation.

Reply:
@Future-Genius this comment has yet to be addressed.
> This class looks very similar to CompleteServletAnnotationQuickFix: https://github.com/eclipse/lsp4jakarta/blob/main/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/servlet/CompleteServletAnnotationQuickFix.java
> Can we create a common class which takes in the attribute names? Then both classes can extend the common class.

Reply:
This one I think can be marked as resolved